packages/common/src/directives/ng_component_outlet.ts

```
@@ -87,45 +98,99 @@ export class NgComponentOutlet implements OnChanges, OnDestroy {
-        } else {
-          this._moduleRef = undefined;
+        if (ngModuleChange || ngModuleFactoryChange) {
+          this._moduleRef?.destroy();
+
+          if (ngModule) {
+            this._moduleRef = createNgModule(ngModule, getParentInjector(injector));
+          } else if (ngModuleFactory) {
+            this._moduleRef = ngModuleFactory.create(getParentInjector(injector));
+          } else {
+            this._moduleRef = undefined;
+          }
         }
-      }
 
-      this._componentRef = viewContainerRef.createComponent(this.ngComponentOutlet, {
-        index: viewContainerRef.length,
-        injector,
-        ngModuleRef: this._moduleRef,
-        projectableNodes: this.ngComponentOutletContent,
-      });
+        this._componentRef = viewContainerRef.createComponent(componentType, {
+          index: viewContainerRef.length,
+          injector,
+          ngModuleRef: this._moduleRef,
+          projectableNodes: content,
+        });
+
+        if (inputs) {
+          const keys = Object.keys(inputs);
+          this.applyComponentInputs(keys);
+        }
+      }
+    } else if (this._componentRef && inputsChanges) {
+      const changedKeys = changedKeysOf(inputsChanges);
+      this.applyComponentInputs(changedKeys);
     }
   }
 
   /** @nodoc */
   ngOnDestroy() {
-    if (this._moduleRef) this._moduleRef.destroy();
+    this._moduleRef?.destroy();
+  }
+
+  private applyComponentInputs(inputNames: string[]) {
+    if (inputNames.length > 0) {
+      const mirror = reflectComponentType(this.ngComponentOutlet!);
+
+      if (mirror) {
+        const inputs = this.ngComponentOutletInputs ?? {};
+        const templateNames = mirror.inputs.map(input => input.templateName);
+
+        inputNames = inputNames.filter(key => templateNames.includes(key));
+
+        for (const name of inputNames) {
+          this._componentRef!.setInput(name, inputs[name]);
+        }
+      }
+    }
```

```
    for (const [key, value] of Object.entries(this. ngComponentOutletInputs)) {
      this.componentRef.setInput(key, value);
    }
```

Arguably we really should throw an error if there's an input that doesn't match rather than filtering it out.

---

AFAICT the error reporting is already built into `setInput`, so there's no need for each call-site to duplicate that